src/library-authoring/library-access/specs/UserAccessContainer.spec.jsx

```diff
@@ -0,0 +1,167 @@
+      }
+    }
+  });
+
+  it.each([
+    [true, LIBRARY_ACCESS.ADMIN, false],
+    [true, LIBRARY_ACCESS.STAFF, true],
+    [true, LIBRARY_ACCESS.USER, false],
+    [false, LIBRARY_ACCESS.STAFF, false],
+  ])(('Staff management button visibility is %s when user admin status is %s, and target user is %s.'),
+    async (isAdmin, accessLevel, buttonsShown) => {
+      const [library] = libraryFactoryLine();
+      const [currentUser, targetUser] = userFactoryLine([{}, { access_level: accessLevel }]);
+      const props = {
+        library, user: targetUser, multipleAdmins: false, isAdmin, ...commonMocks(),
+      };
+      const { container } = ctxRender(
+        <UserAccessWidgetContainer
+          {...props}
+        />,
+        { authenticatedUser: currentUser },
+      );
+      if (buttonsShown) {
+        const removeButton = getByRole(container, 'button', { name: /Remove Staff/ });
+        expect(removeButton).toBeTruthy();
+        const addButton = getByRole(container, 'button', { name: /Add Admin/ });
+        expect(addButton).toBeTruthy();
+      } else {
+        const buttonList = queryAllByRole(container, 'button', { name: /(Remove Staff|Add Admin)/ });
+        expect(buttonList.length).toBe(0);
+      }
+    });
+
+  it.each([
+    [true, LIBRARY_ACCESS.ADMIN, false],
+    [true, LIBRARY_ACCESS.STAFF, false],
+    [true, LIBRARY_ACCESS.USER, true],
+    [false, LIBRARY_ACCESS.USER, false],
+  ])(('User promote button visibility is %s when user admin status is %s, and target user is %s.'),
+    async (isAdmin, accessLevel, buttonShown) => {
+      const [library] = libraryFactoryLine();
+      const [currentUser, targetUser] = userFactoryLine([{}, { access_level: accessLevel }]);
+      const props = {
+        library, user: targetUser, multipleAdmins: false, isAdmin, ...commonMocks(),
+      };
+      const { container } = ctxRender(
+        <UserAccessWidgetContainer
+          {...props}
+        />,
+        { authenticatedUser: currentUser },
+      );
+      const buttonList = queryAllByRole(container, 'button', { name: /Add Staff/ });
+      if (buttonShown) {
+        expect(buttonList.length).toBe(1);
+      } else {
+        expect(buttonList.length).toBe(0);
+      }
+    });
+
+  it.each([
+    [true, LIBRARY_ACCESS.ADMIN, false, false],
+    [true, LIBRARY_ACCESS.ADMIN, true, true],
+    [true, LIBRARY_ACCESS.STAFF, true, true],
+    [true, LIBRARY_ACCESS.STAFF, false, true],
+    [true, LIBRARY_ACCESS.USER, true, true],
+    [true, LIBRARY_ACCESS.USER, false, true],
+    [false, LIBRARY_ACCESS.ADMIN, true, false],
+    [false, LIBRARY_ACCESS.ADMIN, false, false],
+    [false, LIBRARY_ACCESS.STAFF, true, false],
+    [false, LIBRARY_ACCESS.STAFF, false, false],
+    [false, LIBRARY_ACCESS.USER, true, false],
+    [false, LIBRARY_ACCESS.USER, false, false],
+  ])(('Remove user button visibility is %s when user admin status is %s, target user is %s, and multipleAdmins is %s.'),
```

Comment:
Nice use of data-driven testing to capture a wide range of behavior in a single test function :+1: 

When using this pattern, though, keep in mind the future audience of tests: developers who are trying to figure out why they are failing. Data-driven testing has a tendency to be unfriendly here. `Remove user button visibility is false when user admin status is read, target user is false, and multipleAdmins is false` is not a test case failure I would immediately feel empowered to diagnose.

A few suggestions to improve their legibility:
* Sanity-check the order of the positional arguments in these format strings.
* Play with the format of the description. Maybe not trying to work them into prose would help:  `removeUserButtonVisibility==true when adminStatus==...`. Or maybe there's another approach that I'm not thinking of.
* Add a comment to each case of the test describing the case it's testing in a more human way.
* If you're struggling to coherently describe cases of a test, there's no harm in breaking up one test function into two and repeating some code.

Don't feel like you need to do all of these -- just get them to a point where you feel happy looking at the output of `npm run test -- --verbose`

Reply:
I decided to drop the use of it.each since it neither works cleanly with our linter nor is able to provide useful test names when doing positional only interpolation. Instead, I'm now using a foreach loop on the relevant tests directly, while constructing the test name dynamically.